### PR TITLE
Smart fhir admin enable / disable management

### PR DESCRIPTION
**Pull request by adunsulag**
#4091 #4076
This incorporates my other pull request [https://github.com/openemr/openemr/pull/4141](https://github.com/openemr/openemr/pull/4141) just in case we want to bring the whole thing into 6.0.

I've gone ahead and implemented a superuser admin panel available from the menu Administration -> System -> API Clients.

API clients are listed on the screen with a basic details of each client registration.  You can view each client and see their details.  Top right header on the edit screen has an enable / disable button that lets you turn on and off API clients.
Some screenshots:
List Screen
image

Edit screen
image

Enable client
image

Currently this ONLY turns off and on SMART apps.  API clients are still able to make calls into the system.  I'll need to work with @sjpadgett to find out where best to put into AuthorizationController or dispatch.php to handle the rejection of the client id.  

I also noticed we are not keeping track of the granted access tokens.  I planned on displaying a list of all the granted access tokens for each app and letting you revoke an access token from the edit screen.  However, we don't have that information.  We'll need that for MU3 so that will have to be a 6.1 thing.  
image

### CODE REVIEW

**Comment by stephenwaite** [comment]
88220254-ba0afc00-cc30-11ea-9dd4-503ca1172e0e

**Comment by bradymiller** [comment]
hi @adunsulag ,
This code looks great! Gonna bring this in and then rush in some minor translating/escaping work on it

File: src/Common/Auth/OpenIDConnect/Repositories/ClientRepository.php [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 namespace OpenEMR\Common\Auth\OpenIDConnect\Repositories;
 
+use http\Exception\RuntimeException;
```
Comment (bradymiller) [inline src/Common/Auth/OpenIDConnect/Repositories/ClientRepository.php:15]: @adunsulag , Just confirming that above line is a real thing. Tried to look it up on google and couldn't find it.

Comment (adunsulag) [reply src/Common/Auth/OpenIDConnect/Repositories/ClientRepository.php:15]: @bradymiller Shoot, wrong class.  Should be this one: https://www.php.net/manual/en/class.runtimeexception.php  

```use \RuntimeException```

Not sure why phpstorm auto inserted that one.  Good catch.

**Comment by sjpadgett** [comment]
@adunsulag Very nice job on this. I'm thinking of putting you in for promotion!
I'm currently Senior Vice President of Stuff, so, how about Vice President of Other Stuff?

**Comment by adunsulag** [comment]
@sjpadgett Happy New Year, glad you like it.  I think Imperial Admiral of the Outer Isles Master of All Other Stuff appointed by His Majesty Grand Pumba of Stuff.

**Comment by sjpadgett** [comment]
lol. Quick story. Back around 1985 I was offered a promotion of manager over a field engineering group which I was excited about until they asked, as my first duty, to layoff the top three engineers in the group. Now these guys had been with the company the longest and had families so I turned down the so called promotion. I was laid off the next day!
So, are you sure you want the fancy title? There may be strings attached!:)

**Pull request merged**